grails-app/services/org/pih/warehouse/data/LoadDataService.groovy

```diff
@@ -0,0 +1,99 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/ 
+package org.pih.warehouse.data
+
+import au.com.bytecode.opencsv.CSVReader
+import org.grails.plugins.csv.CSVReaderUtils
+import org.pih.warehouse.core.LocationGroup
+import org.pih.warehouse.core.Organization
+import org.pih.warehouse.core.PartyRole
+import org.pih.warehouse.core.PartyType
+import org.pih.warehouse.core.RoleType
+import org.pih.warehouse.importer.LocationCsvImporter
+
+class LoadDataService {
+
+    def locationDataService
+    def productService
+
+    def importLocations(URL csvURL) {
+        BufferedReader reader = csvURL.newInputStream().newReader();
+
+        LocationCsvImporter importer = new LocationCsvImporter(
+                reader,
+                locationDataService
+        );
+
+
+        importer.importData()
+
+        reader.close();
+    }
+
+
+    def importLocationGroups(URL csvURL) {
+        BufferedReader reader = csvURL.newInputStream().newReader();
+
+        CSVReader csvReader = CSVReaderUtils.toCsvReader(reader, [
+                'skipLines': 1
+        ]);
+
+        int nameIndex = 0;
```

Comment:
@hdsldv I think it would be worth keeping in the `demoData` config and it could be:
`int nameIndex = grailsApplication.config.openboxes.demoData.locationGroups.nameIndex ?: 0`

(side note: you don't have to add semicolons at the end of lines)

Reply:
I changed it to parse column names automatically so I can access column values via names